### PR TITLE
Added the M2M enablement under param control. Will enable it once the…

### PR DESCRIPTION
**Pull request by venkatp-xilinx**
Added the M2M enablement under param control. Will enable it once the M2M kernel change is addressed

### CODE REVIEW

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx [hunk 1]
```diff
@@ -2777,42 +2768,46 @@ int HwEmShim::xclCopyBO(unsigned int dst_boHandle, unsigned int src_boHandle, si
     return -1;
   }
    
-  // Disabling the m2m for timebeing as it is not working as expected. So still data is getting transferred thru DMA.
-  // Will enable this logic unless we have a clarity from the m2m hw_emu kernel. Please do not remove this code 
-  /*if ( deviceQuery(key_type::m2m) && getM2MAddress() != 0 ) {
+  // Enabling the copy buffer thru the M2M under the env control. Once the issues regarding this
+  // is resolved from the sim_m2m kernel, we will remove this env control.
+  if (getenv("ENABLE_M2M_KERNEL")) {
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx:2773]: Why don't you use one more xrt.ini variable to decide. Its better to stop using the environment variables wherever possible.

Comment (venkatp-xilinx) [reply src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx:2773]: This is very temporary and will stay for a week or so, hence went with this short cut. INI option is generally used for enabling or disabling the feature and that to for a long time. This is added for convivence for time being to facilitate the Hemants team to test and fix the issue from model side.

Comment (venkatp-xilinx) [reply src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx]: as suggested, replaced the ENV with the ini param

Comment (chvamshi-xilinx) [reply src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx]: Thanks @venkatp-xilinx

**Pull request merged**